pkg/resource/stack/deployment.go

```diff
@@ -482,19 +501,32 @@ func DeserializePropertyValue(v interface{}, dec config.Decrypter) (resource.Pro
 					contract.Assert(isarchive)
 					return resource.NewArchiveProperty(archive), nil
 				case resource.SecretSig:
-					ciphertext, ok := objmap["ciphertext"].(string)
-					if !ok {
-						return resource.PropertyValue{}, errors.New("malformed secret value: missing ciphertext")
+					ciphertext, ciperOk := objmap["ciphertext"].(string)
+					plaintext, plainOk := objmap["plaintext"].(string)
+					if !ciperOk && !plainOk {
+						return resource.PropertyValue{}, errors.New(
+							"malformed secret value: missing ciphertext or plaintext")
 					}
-					var elem interface{}
-					plaintext, err := dec.DecryptValue(ciphertext)
-					if err != nil {
-						return resource.PropertyValue{}, errors.Wrap(err, "decrypting secret value")
+
+					if plainOk {
+						encryptedText, err := enc.EncryptValue(plaintext)
+						if err != nil {
+							return resource.PropertyValue{}, errors.Wrap(err, "encrypting secret value")
+						}
+						ciphertext = encryptedText
```

Comment:
Why do we actually have to do this encryption and compute this ciphertext?

It seems the only place this is used is to populate the encryption cache - but that shouldn't be needed if we don't even have to call Encrypt in the first place to compute this.

If we could avoid this, it also avoids plumbing an encyrpter through all of this code.

Reply:
The normal stack import expects the text to be still in it's encrypted format - this change allows a stack that has been expected with secrets to be re-imported

Reply:
Can you explain this in a bit more detail? I'm also surprised that this is necessary.

Reply:
```
// If the decrypter is a cachingCrypter, insert the plain- and ciphertext into the cache with the
// new *resource.Secret as the key.
```

basically, without this work, we cannot reconstruct the secret
the existing codepath takes the secret, decrypts it and ensures it's correct
then reconstructs the secret
as the import (with secrets) doesn't have the secret, we need to re-encrypt the secret to be able to recreate the secret struct